CONTRIBUTING.md

```diff
@@ -1,8 +1,90 @@
-<!-- TODO: Section on nix CI + devShell -->
+# Contributing guide
 
-## Running tests locally
+Contributions are more than welcome!
 
-For reproducibility, we only run tests that can be sandboxed in CI.
+## Commit messages / PR title
+
+Please ensure your pull request title conforms to [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
+
+## CI
+
+Our CI checks are run using [`nix`](https://nixos.org/download.html#download-nix).
+
+## Development
+
+To be able to reproduce CI, we recommend using `nix`.
+
+### Dev environment
+
+We use the following tools:
+
+#### Formatting
+
+- [`rustfmt`](https://github.com/rust-lang/rustfmt) [Rust]
+- [`alejandra`](https://github.com/kamadorueda/alejandra) [Nix]
+
+#### Linting
+
+- [`cargo check`](https://doc.rust-lang.org/cargo/commands/cargo-check.html)
+- [`clippy`](https://doc.rust-lang.org/clippy/)
+
```

Comment:
I am not sure, are we using "nix" xD ?! joke apart I feel it could be less verbose by saying `we use nix with the (widely used) experimental feature "flake"`.
Also this  listing those can get ouf of date maybe faster than just pointing at how to find them. I guess they are listed in a precommit ? for instance we might want to replace alejandra by nixpkgs-fmt which is going to be used for nixpkgs.
And this misses the lsp lua server.

Reply:
for sure it's better, more work though :)

Reply:
I've added a comment to the relevant location in flake.nix, so we're reminded to update `CONTRIBUTING.md` if we add/remove hooks.